### PR TITLE
sdk updates eric

### PR DESCRIPTION
**Pull request by egill31**

### CODE REVIEW

File: gigapipe/api_client/api/clusters.py [hunk 1]
```diff
@@ -344,3 +345,76 @@ def delete_cluster(self, cluster_slug: str) -> Response:
             )
         return response
 
+    @GigapipeApi.autorefresh_access_token
+    def change_clickhouse_version(self, cluster_slug: str, payload: Dict[str, int]) -> Response:
```
Comment (pausanchezv) [inline gigapipe/api_client/api/clusters.py:349]: Please, force to add the param names after cluster_slug. You have to add an extra param '*'

File: gigapipe/api_client/api/clusters.py [hunk 2]
```diff
@@ -344,3 +345,76 @@ def delete_cluster(self, cluster_slug: str) -> Response:
...
+        except requests.RequestException as e:
+            raise GigapipeServerError(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                message=f"Internal Server Error: {e}"
+            )
+        return response
+
+    @GigapipeApi.autorefresh_access_token
+    def engine_to_replicated_merge_tree(
+            self, cluster_slug: str, table_name: str, payload: Dict[str, str]
```
Comment (pausanchezv) [inline gigapipe/api_client/api/clusters.py:371]: Please, force to add the param names after cluster_slug. You have to add an extra param '*'

File: gigapipe/api_client/api/clusters.py [hunk 3]
```diff
@@ -344,3 +345,76 @@ def delete_cluster(self, cluster_slug: str) -> Response:
...
+        except requests.RequestException as e:
+            raise GigapipeServerError(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                message=f"Internal Server Error: {e}"
+            )
+        return response
+
+    @GigapipeApi.autorefresh_access_token
+    def import_from_external_cluster(
+            self, cluster_slug: str, external_cluster_params: Dict[str, str], table_arrays: List[Dict[str, Any]]
```
Comment (pausanchezv) [inline gigapipe/api_client/api/clusters.py:396]: Please, force to add the param names after cluster_slug. You have to add an extra param '*'

File: gigapipe/api_client/gigapipe_client.py [hunk 4]
```diff
@@ -17,7 +17,7 @@ class GigapipeClient(object):
     Gigapipe Python Client
     """
 
-    gigapipe_url: str = "https://api.gigapipe.com"
```
Comment (pausanchezv) [inline gigapipe/api_client/gigapipe_client.py:20]: Please change the url to prod

File: README.md [hunk 5]
```diff
@@ -675,6 +675,61 @@ gigapipe_client.clusters.delete_cluster(cluster_slug="cluster-test")
...
+    "message": "Changing ClickHouse version in cluster 'cluster-test'...",
+    ...
+}
+```
+
+- Change table engine to ReplicatedMergeTree type
+> Given a cluster slug and a table name, change a table engine to type ReplicatedMergeTree
+```python
+gigapipe_client.clusters.engine_to_replicated_merge_tree(
+    cluster_slug="cluster-test",
```
Comment (pausanchezv) [inline README.md:694]: Please follow the way, the SDK docs are done, where we never use the cluster_slug name param. The rest of named params is totally fine. It is not that one thing is better than the other. It is just to follow the same guidelines. Thanks :)
(Ha a look at the rest of the docs for a better insight)

File: README.md [hunk 6]
```diff
@@ -675,6 +675,61 @@ gigapipe_client.clusters.delete_cluster(cluster_slug="cluster-test")
...
+{
+    "message": "Table engine changed to replicated merge tree: <>"
+}
+```
+
+- Import data from an external cluster, creating tables if necessary
+> Given a cluster slug, external connection parameters and a list of tables, import data from an external cluster.
+```python
+gigapipe_client.clusters.import_from_external_cluster(
+    cluster_slug="cluster-test",
```
Comment (pausanchezv) [inline README.md:713]: Please follow the way, the SDK docs are done, where we never use the cluster_slug name param. The rest of named params is totally fine. It is not that one thing is better than the other. It is just to follow the same guidelines. Thanks :)
(Ha a look at the rest of the docs for a better insight)

File: README.md [hunk 7]
```diff
@@ -675,6 +675,61 @@ gigapipe_client.clusters.delete_cluster(cluster_slug="cluster-test")
 }
 ```
 
+- Change ClickHouse version of Cluster
+> Given a slug and a ClickHouse version id, this method changes the ClickHouse version of a cluster
+```python
+gigapipe_client.clusters.change_clickhouse_version(cluster_slug="cluster-test", payload={"id": 1})
```
Comment (pausanchezv) [inline README.md:681]: Please follow the way, the SDK docs are done, where we never use the cluster_slug name param. The rest of named params is totally fine. It is not that one thing is better than the other. It is just to follow the same guidelines. Thanks :)
(Ha a look at the rest of the docs for a better insight)

**Review by pausanchezv — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (egill31) [inline README.md:681]:
```suggestion
gigapipe_client.clusters.change_clickhouse_version("cluster-test", payload={"id": 1})
```

File: README.md (hunk 5 above)
Comment (egill31) [inline README.md:694]:
```suggestion
    "cluster-test", 
```

File: README.md (hunk 6 above)
Comment (egill31) [inline README.md:713]:
```suggestion
    "cluster-test",
```

File: gigapipe/api_client/api/clusters.py (hunk 1 above)
Comment (egill31) [inline gigapipe/api_client/api/clusters.py:349]:
```suggestion
    def change_clickhouse_version(self, cluster_slug: str, *, payload: Dict[str, int]) -> Response:
```

File: gigapipe/api_client/api/clusters.py (hunk 2 above)
Comment (egill31) [inline gigapipe/api_client/api/clusters.py:371]:
```suggestion
            self, cluster_slug: str, table_name: str, *, payload: Dict[str, str]
```

File: gigapipe/api_client/api/clusters.py (hunk 3 above)
Comment (egill31) [inline gigapipe/api_client/api/clusters.py:396]:
```suggestion
            self, cluster_slug: str, *, external_cluster_params: Dict[str, str], table_arrays: List[Dict[str, Any]]
```

File: gigapipe/api_client/gigapipe_client.py [hunk 8]
```diff
@@ -17,7 +17,7 @@ class GigapipeClient(object):
...
+    gigapipe_url: str = "http://localhost:8000"
```
Comment (egill31) [inline gigapipe/api_client/gigapipe_client.py:20]:
```suggestion
    gigapipe_url: str = "https://api.gigapipe.com"
```

**Pull request merged**